### DISCUSSION TITLE
同时生成静态库和动态库

### DISCUSSION THREAD

**Discussion by xixioba** [xmake-io/xmake#3749] [Q&A]
[target:set_kind](https://xmake.io/#/manual/project_target?id=targetset_kind)一次只能生成静态库或者动态库
我在使用MSVC时，遇到了一些问题（我不确定是不是MSVC问题），例如我有A.dll动态库，编译B.dll时希望依赖A.dll，但xmake命令行尝试使用了A.lib，但我没有办法同时生成同名的静态库
我做的修补是设定不同的target以便同时生成，但这会导致库名字不一致（通过魔改可以解决一部分问题）
所以最终有两个问题：
1. 能否添加同时生成静态库和动态库的功能，如`set_kind("library")`
2. msvc编译dll时是否依赖库会强制使用.lib，如何修改呢

**Comment by waruqi** [comment]
生成 A.dll ，只要你导出了符号。。编译器会自动在对应 dll 目录下生成一个 A.lib 导入库，用于 link A.dll 的。。你无法在当前目录，再去生成一个静态库 A.lib 。。当然会冲突。。

> 能否添加同时生成静态库和动态库的功能，如set_kind("library")

定义两个 target, 一个 static 一个 shared

**Reply by xixioba** [reply]
OK，windows平台太久不用都忘了，我是想测试移植windows下dll和.a的链接运行是否正常，这么看我只能魔改了
有办法拿到`add_deps`的属性然后手动修改么？比如#2841中的`find_package`
